File: src/lessons/welcome/bat/bool2/LastDigit2Entity.java
package lessons.welcome.bat.bool2;

import static plm.core.ValueSerializer.deserialize;
import static plm.core.ValueSerializer.serialize;

import plm.universe.bat.BatEntity;

public class LastDigit2Entity extends BatEntity {

  public void run()
  {
    int count = getTestCount();
    for (int i = 0; i < count; i++) {
      Object[] param = (Object[])deserialize(getTest(i));
      setTestResult(i, serialize(lastDigit((Integer)param[0], (Integer)param[1], (Integer)param[2])));
    }
  }

  /* BEGIN TEMPLATE */
  boolean lastDigit(int a, int b, int c)
  {
    /* BEGIN SOLUTION */
    int da = a % 10;
    int db = b % 10;
    int dc = c % 10;
    return da == db || da == dc || dc == db;
    /* END SOLUTION */
  }
  /* END TEMPLATE */
}

File: src/lessons/welcome/bat/bool2/LastDigit2Entity.py
from RemoteBat import *
def run():
    count = getTestCount()
    for i in range(count):
        param = deserialize(getTest(i))
        setTestResult(i, serialize(lastDigit(param[0], param[1], param[2])))

# BEGIN TEMPLATE
def lastDigit(a, b, c):
# BEGIN SOLUTION
	da = a % 10
	db = b % 10
	dc = c % 10
	return da == db or da == dc or dc == db
# END SOLUTION
# END TEMPLATE
